server/fw_util_iptables.c

```diff
@@ -1498,6 +1583,9 @@ process_spa_request(const fko_srv_options_t * const opts,
     time_t          now;
     unsigned int    exp_ts;
 
+    /* XXX set adequately per SPA message */
+    int             ipv6 = (opts->family == AF_INET6) ? 1 : 0;
+
```

Comment:
Something like
```c
    int ipv6 = 0;
    if(is_valid_ip_addr(spadat->use_src_ip, strlen(spadat->use_src_ip), AF_INET6))
    {
        ipv6 = 1;
    }
```
works to distinguish IPv4 and IPv6 in the SPA message.
It was already checked in `incoming_spa` than the IP is valid with `is_valid_ip_addr(…, AF_UNSPEC)` so there is no security issue not to do the same check for IPv4.

With this change, both IPv4 and IPv6 are handled by fwknopd (which solves my functional issue 2).